src/formatDate.js

```diff
@@ -50,7 +50,29 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const oldDate = date.split(fromSeparator);
+  const newDate = [];
+  const dateInfo = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateInfo[fromFormat[i]] = oldDate[i];
+
+    if (fromFormat[i] === 'YY') {
+      dateInfo.YYYY = (dateInfo.YY < 30 ? '20' : '19') + dateInfo.YY;
```

Comment:
```suggestion
      dateInfo.YYYY = (dateInfo.YY < 30 ? '20' : '19') + dateInfo.YY;
```
`dateInfo.YY` can possibly be undefined. Use the oldDate array instead. In both cases.